src/assets/index.js

```diff
@@ -118,16 +118,39 @@ const showImage = event => {
 /* Getting the query from the url */
 const queryString = window.location.search;
 const urlParams = new URLSearchParams(queryString);
-const SectionToBeDisplay = urlParams.get('q')
-console.log(SectionToBeDisplay);
+let SectionToBeDisplay = urlParams.get('q')
+
+SectionToBeDisplay = SectionToBeDisplay || 'landing';
+// console.log(SectionToBeDisplay);
 
 
-if (SectionToBeDisplay=='analyze')
-{
+if (SectionToBeDisplay=='analyze'){
+    window.history.pushState("Analyze Page", "Crop AI", '?q=analyze');
     show_input_container();
 }
-else if (SectionToBeDisplay=='result')
-{
-    console.log('redirecting to home page')
-    window.location = "index.html";
+
+else if (SectionToBeDisplay=='result'){
+    window.history.pushState("Result Page", "Crop AI", '?q=result');
+    show_result_container();
+}
+ 
+else {
+    window.history.pushState("Landing Page", "Crop AI", 'index.html');
+    show_landing_container();
 }
+    
+window.addEventListener('popstate', () => {
+    
+    console.log(SectionToBeDisplay);
+    window.location.reload();
+    if (SectionToBeDisplay === 'analyze') {
+        window.history.pushState("Landing Page", "Crop AI", 'index.html');
+        show_landing_container();
+    }
+    
```

Comment:
I think this **if condition** is not required
I don't think it is ever executed, as the page refreshed before that